test/framework.cc

```diff
@@ -137,14 +144,40 @@ TestHttpContext::Result TestHttpContext::SendResponseHeaders(
   return std::move(result_);
 }
 
-absl::Status HttpTest::CreatePlugin(const std::string& engine,
-                                    const std::string& wasm_path,
-                                    const std::string& plugin_config) {
+namespace {
+std::string ReadDataFile(const std::string& path) {
+  std::ifstream file(path, std::ios::binary);
+  EXPECT_FALSE(file.fail()) << "failed to open: " << path;
+  std::stringstream file_string_stream;
+  file_string_stream << file.rdbuf();
+  return file_string_stream.str();
+}
+}  // namespace
+
+std::vector<std::string> FindPlugins() {
+  std::vector<std::string> out;
+  for (const auto& entry : std::filesystem::recursive_directory_iterator(".")) {
```

Comment:
tbh I'm surprised this works — why doesn't this recurse through all of bazel-bin, bazel-out, bazel-service-extensions-sampes, bazel-testlogs, as well as the other files in the repository?

Reply:
Bazel is nice -- it organizes every binary into its own execution directory with all its data dependencies, and runs it from there (using symlinks to prevent file duplication). For example the add_header test runs from: .../execroot/__main__/bazel-out/k8-opt/bin/samples/add_header/plugin_test.runfiles/__main__/

But I see your point -- if I run the benchmark from bazel-bin/ directly then things break. 

Looks like I should search from the executable path, and there are various tools for this (whereami, boost::dll::program_location, etc). I'll see what works best.

Reply:
Used boost to scan for files next to the running binary. PTAL.

Reply:
That seems like it will work well with data dependencies. LGTM.